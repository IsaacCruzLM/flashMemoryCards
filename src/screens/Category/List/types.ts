import {ReactNode} from 'react';
import {CategoryModelType} from '../../../databases/models/categoryModel';
import {NoteModelType} from '../../../databases/models/noteModel';

export interface ListProps {
  categories: Array<CategoryModelType>;
  notes: Array<NoteModelType>;
  children?: ReactNode;
}
