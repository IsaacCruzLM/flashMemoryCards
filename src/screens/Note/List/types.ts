import {DefaultFormProps} from '../../../components/Form/types';

import {NoteModelType} from '../../../databases/models/noteModel';
import {NoteSubjectModelType} from '../../../databases/models/noteSubjectModel';
import {CategoryModelType} from '../../../databases/models/categoryModel';
import {SubjectModelType} from '../../../databases/models/subjectModel';

export interface ListProps {
  route?: object;
  notes?: NoteModelType[];
  category?: CategoryModelType;
  noteSubjects?: NoteSubjectModelType[];
  categories: CategoryModelType[];
  subjects: SubjectModelType[];
}

export interface NewCategoryFormProps extends DefaultFormProps {
  values: formValues;
}

export interface formValues {
  nome: string;
  icon: string;
  color: string;
}

export interface sectionData {
  title: string;
  data: Array<CardData>;
}

export interface CardData {
  id: string;
  title: string;
  creationDate: string;
  lastRevisionDate: string;
  noteType: string;
  category: string;
  subjects: Array<subjectData>;
}

export interface subjectData {
  content: string;
  color: string;
}

export interface filterState {
  category: string;
  subjects: string[];
}

export interface filterProps extends ListProps {
  filters: filterState;
  handleFilterChange: Function;
}
