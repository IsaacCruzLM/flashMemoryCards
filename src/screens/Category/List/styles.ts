import {StyleSheet} from 'react-native';
import themes from '../../../styles/themes';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    width: '100%',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  listContainer: {
    paddingVertical: themes.spacing.unit * 3,
    paddingHorizontal: themes.spacing.unit * 2 + 4,
  },
  flatList: {
    flex: 1,
    width: '100%',
  },
});

export default styles;
