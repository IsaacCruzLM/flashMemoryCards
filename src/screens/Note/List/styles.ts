import {StyleSheet} from 'react-native';
import themes from '../../../styles/themes';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    width: '100%',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  formContainer: {
    width: '100%',
  },
  createButton: {
    marginBottom: themes.spacing.unit * 3,
  },
  listHeader: {
    fontWeight: '700',
    fontFamily: themes.fonts.medium.fontFamily,
    fontSize: themes.typography.fontSizeLargeTitle,
    color: themes.colors.primary,
    marginBottom: themes.spacing.unit * 2,
  },
  sectionList: {
    flex: 1,
    width: '100%',
  },
  cardCustomStyle: {
    marginBottom: themes.spacing.unit * 1.5,
  },
  listContainer: {
    paddingVertical: themes.spacing.unit * 3,
    paddingHorizontal: themes.spacing.unit * 2 + 4,
  },
});

export default styles;
