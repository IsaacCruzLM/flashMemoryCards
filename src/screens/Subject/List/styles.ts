import {StyleSheet} from 'react-native';
import themes from '../../../styles/themes';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    width: '100%',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  listContainer: {
    paddingVertical: themes.spacing.unit * 3,
    paddingHorizontal: themes.spacing.unit * 2 + 4,
  },
  flatList: {
    flex: 1,
    width: '100%',
  },
  itemStyle: {
    marginBottom: themes.spacing.unit * 1.5,
  },
});

export default styles;
