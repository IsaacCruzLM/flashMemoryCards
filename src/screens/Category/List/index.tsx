import React, {useContext} from 'react';
import {FlatList} from 'react-native';
import withObservables from '@nozbe/with-observables';
import {withDatabase} from '@nozbe/watermelondb/DatabaseProvider';
import {compose} from 'recompose';

import AppContext from '../../../context/appContext';
import NavigationService from '../../../navigation/NavigationService';

import CategoryListCard from '../../../components/CategoryListCard';
import FloatingAddButton from '../../../components/FloatingAddButton';
import EmpytMessage from '../../../components/EmpytMessage';

import styles from './styles';

const List = ({categories}: any) => {
  const {setCurrentCategoryName} = useContext(AppContext);

  if (categories.length <= 0) {
    return (
      <EmpytMessage
        message={'Nenhuma Categoria Encontrada'}
        actionLabel={'Crie uma nova categoria'}
        onPressAction={() => NavigationService.navigate('NewCategory')}
      />
    );
  }

  return (
    <>
      <FlatList
        data={categories}
        contentContainerStyle={styles.listContainer}
        keyExtractor={(item, index) => item.title + index}
        renderItem={({item}) => (
          <CategoryListCard
            title={item.name}
            creationDate={item.createdAt.toLocaleDateString('pt-br')}
            numberOfNotes={16}
            icon={item.icon}
            numberNotesToReview={2}
            onPress={() => {
              setCurrentCategoryName(item.name);
              NavigationService.navigate('Notes', {categoryId: item.id});
            }}
            containerStyle={styles.itemStyle}
          />
        )}
        style={styles.flatList}
      />
      <FloatingAddButton routeName="NewCategory" />
    </>
  );
};

export default compose(
  withDatabase,
  withObservables([], ({database}: any) => ({
    categories: database.get('categories').query(),
  })),
)(List);
