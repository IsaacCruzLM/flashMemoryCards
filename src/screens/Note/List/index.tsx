/* eslint-disable react-hooks/exhaustive-deps */
import React, {useEffect, useState, useContext} from 'react';
import {Text, SectionList, SectionListData, View} from 'react-native';
import {withDatabase} from '@nozbe/watermelondb/DatabaseProvider';
import withObservables from '@nozbe/with-observables';
import {Q} from '@nozbe/watermelondb';
import {compose} from 'recompose';
import get from 'lodash/get';
import cloneDeep from 'lodash/cloneDeep';

import NoteListCard from '../../../components/NoteListCard';
import FloatingAddButton from '../../../components/FloatingAddButton';
import EmpytMessage from '../../../components/EmpytMessage';
import Dialog from '../../../components/Dialog';
import Select from '../../../components/Select';
import SelectMultiple from '../../../components/SelectMultiple';
import DataRangeInput from '../../../components/DataRangeInput';
import {rangeDataType} from '../../../components/DataRangeInput/types';

import useGetFromGlobalState from '../../../hooks/useGetFromGlobalState';
import {NoteModelType} from '../../../databases/models/noteModel';
import NavigationService from '../../../navigation/NavigationService';
import noteNeedToBeRevised from '../../../utils/noteValidations';
import noteRevisionUpdate from '../../../utils/noteValidations/noteRevisionUpdate';
import filterActions from '../../../utils/filters';
import AppContext from '../../../context/appContext';
import {translateOptions} from '../Create';

import styles from './styles';
import {
  ListProps,
  CardData,
  sectionData,
  filterProps,
  filterState,
} from './types';

export const NotesFilter: React.FunctionComponent<filterProps> = ({
  filters,
  handleFilterChange,
  categories,
  subjects,
  dontShowCategory,
}) => {
  return (
    <View>
      {dontShowCategory ? null : (
        <Select
          options={translateOptions(categories)}
          onChange={value => handleFilterChange('category', value)}
          modalTitle="Selecione uma categoria"
          inputLabel="Selecionar Categoria"
          inputPlaceHolder="Selecionar Categoria"
          defaultValue={filters.category}
        />
      )}
      <SelectMultiple
        options={translateOptions(subjects as any)}
        onChange={value => handleFilterChange('subjects', value)}
        modalTitle="Selecione vários assuntos"
        inputPlaceHolder="Selecionar Assuntos"
        defaultValue={filters.subjects}
      />
      <DataRangeInput
        label={'Data de criação'}
        onChangeRange={date => handleFilterChange('creationDate', date)}
        defaultValue={filters.creationDate}
      />
      <DataRangeInput
        label={'Ultima revisão'}
        onChangeRange={date => handleFilterChange('lastRevision', date)}
        defaultValue={filters.lastRevision}
      />
      <DataRangeInput
        label={'Próxima revisão'}
        onChangeRange={date => handleFilterChange('nextRevision', date)}
        defaultValue={filters.nextRevision}
      />
    </View>
  );
};

export const DEFAULT_FILTER_STATE = {
  category: '',
  subjects: [],
  creationDate: {init: null, end: null},
  lastRevision: {init: null, end: null},
  nextRevision: {init: null, end: null},
};

const List: React.FunctionComponent<ListProps | any> = ({
  route,
  notes,
  category,
  noteSubjects,
  categories,
  subjects,
}) => {
  const {setFilterDialogOpenFunction} = useContext(AppContext);
  const searchQuery = useGetFromGlobalState('searchParams.Notes', '');
  const openFilterDialog = useGetFromGlobalState(
    'filterDialogOpen.Notes',
    false,
  ) as boolean;

  const [noteBySections, setNoteBySections] = useState(
    [] as SectionListData<any, sectionData>[],
  );
  const [filters, setFilters] = useState(DEFAULT_FILTER_STATE as filterState);

  const filterData = (note: NoteModelType) => {
    const {
      creationDate,
      lastRevision,
      nextRevision,
      category: categoryFilter,
      subjects: subjectsFilter,
    } = cloneDeep(filters);
    let dontFilter = true;

    if (categoryFilter && dontFilter) {
      dontFilter = note.category.id === categoryFilter;
    }

    if (subjectsFilter.length > 0 && dontFilter) {
      const subjectsIds = noteSubjects
        .filter((record: any) => record.note.id === note.id)
        .map((record: any) => record.subject.id);
      const someIdIsIncluded = subjectsIds.some((id: string) =>
        subjectsFilter.includes(id),
      );
      dontFilter = someIdIsIncluded;
    }

    if ((creationDate.init || creationDate.end) && dontFilter) {
      dontFilter = filterActions.rangeInputDateVerify(
        creationDate,
        note.createdAt.toString(),
      );
    }

    if ((lastRevision.init || lastRevision.end) && dontFilter) {
      dontFilter = filterActions.rangeInputDateVerify(
        lastRevision,
        note.lastRevision.toString(),
      );
    }

    if ((nextRevision.init || nextRevision.end) && dontFilter) {
      dontFilter = filterActions.rangeInputDateVerify(
        nextRevision,
        note.nextRevision.toString(),
      );
    }

    return dontFilter;
  };

  useEffect(() => {
    const fetchNotes = async () => {
      if (notes.length > 0) {
        const newNoteBySections = [];
        const otherSection = {
          title: 'Demais anotações',
          data: [] as Array<CardData>,
        };
        const revisionSection = {
          title: 'Anotações a serem revisadas',
          data: [] as Array<CardData>,
        };
        await Promise.all(
          notes.filter(filterData).map(async (note: NoteModelType) => {
            const categoryName = get(category, 'name', '');
            const subjectsFetch = await note.subjects.fetch();
            const needRevision = noteNeedToBeRevised(note);
            if (note.name.includes(searchQuery)) {
              (needRevision ? revisionSection : otherSection).data.push({
                id: note.id,
                title: note.name,
                creationDate: new Date(note.createdAt).toLocaleDateString(
                  'pt-BR',
                ),
                lastRevisionDate: new Date(
                  note.lastRevision,
                ).toLocaleDateString('pt-BR'),
                nextRevisionDate: new Date(
                  note.nextRevision,
                ).toLocaleDateString('pt-BR'),
                noteType: 'Texto',
                category: categoryName,
                subjects: (subjectsFetch || []).map(({name, color}: any) => ({
                  content: name,
                  color,
                })),
                levelRevision: note.levelRevision,
                needRevision: needRevision,
              });
            }
          }),
        );

        if (revisionSection.data.length > 0) {
          newNoteBySections.push(revisionSection);
        }
        if (otherSection.data.length > 0) {
          newNoteBySections.push(otherSection);
        }

        setNoteBySections(newNoteBySections);
      }
    };

    fetchNotes();
  }, [category, notes, noteSubjects, searchQuery, filters]);

  const handleFilterChange = (
    key:
      | 'category'
      | 'subjects'
      | 'creationDate'
      | 'lastRevision'
      | 'nextRevision',
    value: string & string[] & rangeDataType,
  ) => {
    const newFiltersClone = cloneDeep(filters);
    newFiltersClone[key] = value;
    setFilters(newFiltersClone);
  };

  const hideFilterDialog = () => setFilterDialogOpenFunction({Notes: false});

  const resetFilterDialog = () => {
    setFilters(DEFAULT_FILTER_STATE as filterState);
    hideFilterDialog();
  };

  if (notes.length <= 0) {
    return (
      <EmpytMessage
        message={'Nenhuma anotação criada para esta categoria'}
        actionLabel={'Crie uma anotação'}
        onPressAction={() =>
          NavigationService.navigate('NewNote', {
            categoryId: get(route, 'params.categoryId', ''),
          })
        }
      />
    );
  }

  return (
    <>
      <SectionList
        sections={noteBySections}
        contentContainerStyle={styles.listContainer}
        keyExtractor={item => item.id}
        renderItem={({item}) => (
          <NoteListCard
            title={item.title}
            creationDate={item.creationDate}
            lastRevisionDate={item.lastRevisionDate}
            nextRevisionDate={item.nextRevisionDate}
            noteType={item.noteType}
            category={item.category}
            subjects={item.subjects}
            containerStyle={styles.cardCustomStyle}
            onPress={() => {
              noteRevisionUpdate(
                item.id,
                item.levelRevision,
                item.creationDate,
                item.needRevision,
              );
              NavigationService.navigate('ShowNote', {
                noteName: item.title,
                noteId: item.id,
              });
            }}
          />
        )}
        renderSectionHeader={({section: {title}}) =>
          title ? <Text style={styles.listHeader}>{title}</Text> : null
        }
        style={styles.sectionList}
      />
      <FloatingAddButton
        routeName="NewNote"
        params={{categoryId: get(route, 'params.categoryId', '')}}
      />
      <Dialog
        actions={[
          {
            label: 'Limpar',
            buttonMode: 'outlined',
            buttonAction: () => resetFilterDialog(),
          },
          {
            label: 'Filtrar',
            buttonMode: 'contained',
            buttonAction: () => hideFilterDialog(),
          },
        ]}
        isVisible={openFilterDialog}
        hideDialog={() => hideFilterDialog()}
        title={'Adicionar filtros'}>
        <NotesFilter
          filters={filters}
          handleFilterChange={handleFilterChange}
          categories={categories}
          subjects={subjects}
          dontShowCategory
        />
      </Dialog>
    </>
  );
};

export default compose(
  withDatabase,
  withObservables(['route'], ({database, route}: any) => {
    const categoryId = get(route, 'params.categoryId', '');
    return {
      category: database.get('categories').findAndObserve(categoryId),
      notes: database
        .get('notes')
        .query(Q.where('category_id', categoryId))
        .observeWithColumns(['name', 'category_id']),
      noteSubjects: database
        .get('note_subjects')
        .query()
        .observeWithColumns(['note_id', 'subject_id']),
      categories: database.get('categories').query(),
      subjects: database.get('subjects').query(),
    };
  }),
)(List);
