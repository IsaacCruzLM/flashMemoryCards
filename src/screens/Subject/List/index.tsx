import React from 'react';
import {FlatList} from 'react-native';
import withObservables from '@nozbe/with-observables';
import {withDatabase} from '@nozbe/watermelondb/DatabaseProvider';
import {compose} from 'recompose';

import NavigationService from '../../../navigation/NavigationService';

import FloatingAddButton from '../../../components/FloatingAddButton';
import EmpytMessage from '../../../components/EmpytMessage';
import SubjectListCard from '../../../components/SubjectListCard';

import styles from './styles';

const List = ({subjects}: any) => {
  if (subjects.length <= 0) {
    return (
      <EmpytMessage
        message={'Nenhum Assunto Encontrado'}
        actionLabel={'Crie um novo assunto'}
        onPressAction={() => NavigationService.navigate('NewSubject')}
      />
    );
  }

  return (
    <>
      <FlatList
        data={subjects}
        contentContainerStyle={styles.listContainer}
        keyExtractor={item => item.id}
        renderItem={({item}) => {
          const {id, name, color} = item;
          return (
            <SubjectListCard
              title={name}
              subjectColor={color}
              onPress={() =>
                NavigationService.navigate('NewSubject', {
                  subjectId: id,
                  isEdit: true,
                })
              }
              containerStyle={styles.itemStyle}
            />
          );
        }}
        style={styles.flatList}
      />
      <FloatingAddButton routeName="NewSubject" />
    </>
  );
};

export default compose(
  withDatabase,
  withObservables([], ({database}: any) => ({
    subjects: database
      .get('subjects')
      .query()
      .observeWithColumns(['name', 'color']),
  })),
)(List);
