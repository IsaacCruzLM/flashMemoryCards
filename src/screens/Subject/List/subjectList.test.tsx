import React from 'react';
import renderer, {ReactTestInstance} from 'react-test-renderer';
import List from './index';

describe('Subject List Screen', () => {
  let tree: renderer.ReactTestRenderer;
  beforeEach(() => {
    jest.clearAllMocks();

    tree = renderer.create(<List />);
  });

  afterEach(() => {
    jest.clearAllMocks();
    jest.useRealTimers();
  });

  it('should render correctly', () => {
    expect(tree.toJSON()).toBeTruthy();
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should render flat list', () => {
    const flatList = tree.root.findByType('RCTScrollView' as React.ElementType);
    expect(flatList).toBeTruthy();
  });

  it('should render should render two subjects', () => {
    const testInstance = tree.root;

    const matchingTexts = testInstance.findAll((node: ReactTestInstance) => {
      return (
        (node.type as string) === 'Text' &&
        (node.children as string[]).some((text: string) =>
          text.includes('Assunto'),
        )
      );
    });

    expect(matchingTexts.length).toBe(2);
  });

  it('should render floating button', () => {
    const testInstance = tree.root;

    const floatingButton = testInstance.findByProps({
      testID: 'floating-add-container',
    });

    expect(floatingButton).toBeTruthy();
  });
});
