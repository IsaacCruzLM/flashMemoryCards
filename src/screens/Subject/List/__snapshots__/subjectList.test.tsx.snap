// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subject List Screen should render correctly 1`] = `
Array [
  <RCTScrollView
    ListEmptyComponent={[Function]}
    contentContainerStyle={
      Object {
        "flexGrow": 1,
        "paddingHorizontal": 20,
        "paddingVertical": 24,
      }
    }
    data={
      Array [
        Object {
          "color": "#ffffff",
          "id": "subject_1",
          "name": "Assunto 1",
        },
        Object {
          "color": "#d11cd5",
          "id": "subject_2",
          "name": "Assunto 2",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "flex": 1,
        "width": "100%",
      }
    }
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Array [
              Object {
                "width": "100%",
              },
              Object {
                "marginBottom": 12,
              },
            ]
          }
        >
          <View
            accessibilityState={
              Object {
                "disabled": true,
              }
            }
            accessible={true}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flexGrow": 1,
                "flexShrink": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "paddingHorizontal": 16,
                  },
                  Object {
                    "paddingVertical": 16,
                  },
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "padding": 8,
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "borderRadius": 15,
                      "height": 30,
                      "width": 30,
                    },
                    Object {
                      "backgroundColor": "#ffffff",
                    },
                  ]
                }
              />
              <Text
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "color": "rgba(28, 27, 31, 1)",
                      "fontFamily": "System",
                      "fontWeight": "400",
                      "letterSpacing": 0,
                    },
                    Object {
                      "writingDirection": "ltr",
                    },
                    Object {
                      "color": "#22272E",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": "700",
                      "marginLeft": 16,
                    },
                  ]
                }
              >
                Assunto 1
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Array [
              Object {
                "width": "100%",
              },
              Object {
                "marginBottom": 12,
              },
            ]
          }
        >
          <View
            accessibilityState={
              Object {
                "disabled": true,
              }
            }
            accessible={true}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flexGrow": 1,
                "flexShrink": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "paddingHorizontal": 16,
                  },
                  Object {
                    "paddingVertical": 16,
                  },
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "padding": 8,
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "borderRadius": 15,
                      "height": 30,
                      "width": 30,
                    },
                    Object {
                      "backgroundColor": "#d11cd5",
                    },
                  ]
                }
              />
              <Text
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "color": "rgba(28, 27, 31, 1)",
                      "fontFamily": "System",
                      "fontWeight": "400",
                      "letterSpacing": 0,
                    },
                    Object {
                      "writingDirection": "ltr",
                    },
                    Object {
                      "color": "#22272E",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": "700",
                      "marginLeft": 16,
                    },
                  ]
                }
              >
                Assunto 2
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>,
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#22272E",
        "borderRadius": 30,
        "bottom": 24,
        "height": 60,
        "justifyContent": "center",
        "position": "absolute",
        "right": 20,
        "width": 60,
      }
    }
    testID="floating-add-container"
  >
    <Icon
      color="#FFFFFF"
      name="plus"
      size={50}
    />
  </View>,
]
`;
